source/docs/software/vision-processing/grip/reading-array-values-published-by-networktables.rst

```diff
@@ -26,33 +26,63 @@ Both of the following examples are extremely simplified programs that just illus
 the robotInit() method so it's only run when the program starts up. In your programs, you would more likely get the values in code
 that is evaluating which direction to aim the robot in a command or a control loop during the autonomous or teleop periods.
 
-Writing a Java program to access the keys
+Writing a program to access the keys
 -----------------------------------------
-.. figure:: images/reading-array-values-published-by-networktables/writing-a-java-program-to-access-the-keys.png
+.. tabs::
 
-The steps to getting the values and, in this program, printing them are:
+   .. code-tab:: java
 
-1.  Declare the table variable that will hold the instance of the subtable that have the values.
-2.  Initialize the subtable instance so that it can be used later for retrieving the values.
-3.  Read the array of values from NetworkTables. In the case of a communicating programs, it's possible that the program producing
-    the output being read here might not yet be available when the robot program starts up. To avoid issues of the data not being
-    ready, a default array of values is supplied. This default value will be returned if the network table key hasn't yet been
-    published. This code just loops forever and reads values and prints them to the console.
+      NetworkTable table;
+      double[] areas;
+      double[] defaultValue = new double[0];
+
+      @Override
+      public void robotInit() {
+        table = NetworkTableInstance.getDefault().getTable("GRIP/mycontoursReport");
+      }
+
+      @Override
+      public void teleopPeriodic() {
+          double[] areas = table.getEntry("area").getDoubleArray(defaultValue);
+
+          System.out.print("areas: " );
+
+          for (double area : areas) {
+            System.out.print(area + " ");
+          }
 
-Writing a C++ program to access the keys
-----------------------------------------
-.. figure:: images/reading-array-values-published-by-networktables/writing-a-cpp-program-to-access-the-keys.png
+          System.out.println();
+      }
+
+   .. code-tab:: cpp
+
+      std::shared_ptr<NetworkTable> table;
+
+      void Robot::RobotInit()
```

Comment:
For consistency C++ examples should be shown as if it's in the class body as well (like Java), unless we are going to start showing separate .h and .cpp files.